http-client/src/main/java/io/micronaut/http/client/netty/FullNettyClientHttpResponse.java

```diff
@@ -264,20 +217,9 @@ private <T> Optional convertByteBuf(ByteBuf content, Argument<T> type) {
         }
         Optional<MediaType> contentType = getContentType();
         if (contentType.isPresent()) {
-            if (mediaTypeCodecRegistry != null) {
-                bodyBytes = ByteBufUtil.getBytes(content);
-                if (CharSequence.class.isAssignableFrom(type.getType())) {
-                    Charset charset = contentType.flatMap(MediaType::getCharset).orElse(StandardCharsets.UTF_8);
-                    return Optional.of(new String(bodyBytes, charset));
-                } else if (type.getType() == byte[].class) {
-                    return Optional.of(bodyBytes);
-                } else {
-                    Optional<MediaTypeCodec> foundCodec = mediaTypeCodecRegistry.findCodec(contentType.get());
-                    if (foundCodec.isPresent()) {
-                        MediaTypeCodec codec = foundCodec.get();
-                        return Optional.of(codec.decode(type, bodyBytes));
-                    }
-                }
+            Optional<MessageBodyReader<T>> reader = handlerRegistry.findReader(type, List.of(contentType.get()));
+            if (reader.isPresent() && reader.get().isReadable(type, contentType.get())) {
```

Comment:
split into 2 `if` blocks to avoid the multiple calls to `get` also a extract a variable for `contentType` to avoid multiple calls